biojava-core/src/main/java/org/biojava/nbio/core/search/io/blast/BlastXMLQuery.java

```diff
@@ -0,0 +1,270 @@
+                                .setHspGaps(new Integer(XMLHelper.selectSingleElement(hspElement, "Hsp_gaps").getTextContent()))
+                                .setHspAlignLen(new Integer(XMLHelper.selectSingleElement(hspElement, "Hsp_align-len").getTextContent()))
+                                .setHspQseq(XMLHelper.selectSingleElement(hspElement, "Hsp_qseq").getTextContent())
+                                .setHspHseq(XMLHelper.selectSingleElement(hspElement, "Hsp_hseq").getTextContent())
+                                .setHspIdentityString(XMLHelper.selectSingleElement(hspElement, "Hsp_midline").getTextContent());
+
+                            hspsCollection.add(blastHspBuilder.createBlastHsp());
+                        }
+                    }
+                    // finally set the computed hsp collection and create Hit object
+                    blastHitBuilder.setHsps(hspsCollection);
+                    hitsCollection.add(blastHitBuilder.createBlastHit());
+                }
+                // finally set the computed Hit collection to the result
+                resultBuilder.setHits(hitsCollection);
+                resultsCollection.add(resultBuilder.createBlastResult());
+            }
+        } catch (XPathException e) {
+            throw new ParseException(e.getMessage(),0);
+        }
+        log.info("Parsing finished.");
+        
+        return resultsCollection;
+    }
+    
+    public List<String> getFileExtensions(){
+        ArrayList<String> extensions = new ArrayList(1);
+        extensions.add("blastxml");
+        return extensions;
+    }
+
+    @Override
+    public void setQueryReferences(List sequences) {
+        queryReferences = sequences;
+    }
+
+    @Override
+    public void setDatabaseReferences(List sequences) {
+        databaseReferences = sequences;
+    }
+    
+    /**
+     * fill the map association between sequences an a unique id
+     */
+    private void mapIds() {
+        if (queryReferences != null) {
+            queryReferencesMap = new HashMap(queryReferences.size());
+            for (int counter=0; counter < queryReferences.size() ; counter ++){
+                String id = "Query_"+(counter+1);
+                queryReferencesMap.put(id, queryReferences.get(counter));
+            }  
+        }
+        
+        if (databaseReferences != null) {
+            databaseReferencesMap = new HashMap(databaseReferences.size());
+            for (int counter=0; counter < databaseReferences.size() ; counter ++){
+                // this is strange: while Query_id are 1 based, Hit (database) id are 0 based
+                String id = "gnl|BL_ORD_ID|"+(counter);
+                databaseReferencesMap.put(id, databaseReferences.get(counter));
+            }  
+        }
+    }
+
+    @Override
+    public void storeObjects(List<Result> results) throws Exception {
+        throw new UnsupportedOperationException("This parser does not support writing yet.");
+    }
+}

```

Comment:
No need to declare `throws Exception` here, `UnsupportedOperationException` is a runtime exception.